sdk/src/azure/core/az_http_request.c

```diff
@@ -237,14 +147,30 @@ az_http_request_set_query_parameter(az_http_request* ref_request, az_span name,
 
   url_remainder = az_span_copy_u8(url_remainder, separator);
 
+  int32_t encoding_size = 0;
   // Append parameter name
-  url_remainder = az_span_copy(url_remainder, name);
+  if (is_query_url_encoded)
```

Comment:
Do we have to keep checking this or can we split this up so we only need to check it once in the beginning and then go to two separate code paths?

Reply:
it's either repeating the check, or repeating 
`url_remainder = az_span_copy_u8(url_remainder, '=');` 
of both paths

Or creating a new static function to add the `=` and reuse.

I am fine with any approach. Compiler might end up generating the same think for all.